auth_brute_force/models/res_authentication_attempt.py

```diff
@@ -2,35 +2,80 @@
 # Copyright 2015 GRAP - Sylvain LE GAL
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
 
-from odoo import _, api, fields, models
+import logging
+from odoo import api, fields, models, SUPERUSER_ID
+
+_logger = logging.getLogger(__name__)
 
 
 class ResAuthenticationAttempt(models.Model):
     _name = 'res.authentication.attempt'
     _order = 'attempt_date desc'
 
-    _ATTEMPT_RESULT = [
-        ('successfull', _('Successfull')),
-        ('failed', _('Failed')),
-        ('banned', _('Banned')),
-    ]
-
-    # Column Section
-    attempt_date = fields.Datetime(string='Attempt Date')
+    attempt_date = fields.Datetime(
+        default=fields.Datetime.now,
+    )
     login = fields.Char(string='Tried Login')
     remote = fields.Char(string='Remote ID')
     result = fields.Selection(
-        selection=_ATTEMPT_RESULT, string='Authentication Result')
+        string='Authentication Result',
+        selection=[
+            ('successfull', 'Successful'),
+            ('failed', 'Failed'),
+            ('banned', 'Banned'),
+        ],
+    )
+
+    @api.multi
+    def write(self, vals):
+        result = super(ResAuthenticationAttempt, self).write(vals)
+        self._autoban()
+        return result
 
-    # Custom Section
     @api.model
-    def search_last_failed(self, remote):
+    def create(self, vals):
+        result = super(ResAuthenticationAttempt, self).create(vals)
+        result._autoban()
+        return result
+
+    @api.model
+    def search_last_failed(self, remote, until=None):
+        if not until:
+            until = fields.Datetime.now()
         last_ok = self.search(
-            [('result', '=', 'successfull'), ('remote', '=', remote)],
+            [('result', '=', 'successfull'), ('remote', '=', remote),
+             ("attempt_date", "<=", until)],
```

Comment:
Maybe at least some of the fields used in this search should be setted as `index=True` ? On a medium/big installation running for years there could be potentially thousands (or even tens or hundreds of thousands) of login attempts to search in..

Reply:
In the other hand, I don't see any interest to keep attempts with a date > X monthes. So maybe a cron to clean this table should be great. (I think there is a generic module that does that task for any table in server-tools.)

But Indeed, set an index=True is a good idea !